pkg/bgpv1/manager/manager.go

```diff
@@ -502,6 +494,278 @@ func (m *BGPRouterManager) Stop() {
 		s.Server.Stop()
 	}
 
+	for _, i := range m.BGPInstances {
+		i.Router.Stop()
+	}
+
 	m.Servers = make(LocalASNMap)
+	m.BGPInstances = make(LocalInstanceMap)
 	m.running = false
 }
+
+// ReconcileInstances is a API for configuring the BGP Instances from the
+// desired CiliumBGPNodeConfig resource.
+//
+// ReconcileInstances will evaluate BGP instances to be created, removed and
+// reconciled.
+func (m *BGPRouterManager) ReconcileInstances(ctx context.Context,
+	nodeObj *v2alpha1api.CiliumBGPNodeConfig,
+	ciliumNode *v2api.CiliumNode) error {
+	m.Lock()
+	defer m.Unlock()
+
+	l := log.WithFields(
+		logrus.Fields{
+			"component": "manager.v2.ReconcileInstances",
+		},
+	)
+
+	// use a reconcileDiff to compute which BgpServers must be created, removed
+	// and reconciled.
+	rd := newReconcileDiffV2(ciliumNode)
+
+	if nodeObj == nil {
+		return m.withdrawAllV2(ctx, rd)
+	}
+
+	err := rd.diff(m.BGPInstances, nodeObj)
+	if err != nil {
+		return err
+	}
+
+	if rd.empty() {
+		l.Debug("BGP instance up-to-date with CiliumBGPNodeConfig")
+		return nil
+	}
+	l.WithField("diff", rd.String()).Debug("Reconciling BGP instances")
+
+	if len(rd.register) > 0 {
+		if err := m.registerV2(ctx, rd); err != nil {
+			return fmt.Errorf("encountered error adding new BGP Servers: %v", err)
+		}
+	}
+	if len(rd.withdraw) > 0 {
+		if err := m.withdrawV2(ctx, rd); err != nil {
+			return fmt.Errorf("encountered error removing existing BGP Servers: %v", err)
+		}
+	}
+	if len(rd.reconcile) > 0 {
+		if err := m.reconcileV2(ctx, rd); err != nil {
+			return fmt.Errorf("encountered error reconciling existing BGP Servers: %v", err)
+		}
+	}
+
+	return nil
+}
+
+// registerV2 instantiates and configures BGP Instance(s) as instructed by the provided
+// work diff.
+func (m *BGPRouterManager) registerV2(ctx context.Context, rd *reconcileDiffV2) error {
+	l := log.WithFields(
+		logrus.Fields{
+			"component": "manager.v2.register",
+		},
+	)
+	var err error
+	for _, name := range rd.register {
+		var config *v2alpha1api.CiliumBGPNodeInstance
+		var ok bool
+		if config, ok = rd.seen[name]; !ok {
+			l.Errorf("Work diff (add) contains unseen instance %s, skipping", name)
```

Comment:
This is same code as before - this is unlikely to happen as seen map is populated earlier in the call stack. There are no concurrent writes or reads to this map. 

From user prospective, error is not actionable - just opening an issue with the logs so we can investigate why it might have happened.